// angular-tabs-component/__ivy_ngcc__/dist/components/tabs/tab.component.js
import { Component, Input } from "@angular/core";
import { TabsContainer } from "../../containers";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../containers';

var _c0 = ["*"];
var TabComponent = /** @class */ (function () {
    function TabComponent(tabs) {
        tabs.addTab(this);
    }
    TabComponent.prototype.getTabTitle = function () {
        return this.tabTitle;
    };
    /** @nocollapse */
    TabComponent.ctorParameters = function () { return [
        { type: TabsContainer, },
    ]; };
    TabComponent.propDecorators = {
        'active': [{ type: Input },],
        'tabTitle': [{ type: Input },],
    };
TabComponent.ɵfac = function TabComponent_Factory(t) { return new (t || TabComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TabsContainer)); };
TabComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: TabComponent, selectors: [["tab"]], hostVars: 2, hostBindings: function TabComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("hidden", !ctx.active);
    } }, inputs: { active: "active", tabTitle: "tabTitle" }, ngContentSelectors: _c0, decls: 2, vars: 0, consts: [[1, "tabs__panel"]], template: function TabComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } }, styles: ["*[_ngcontent-%COMP%]{margin:0px;padding:0px}[_nghost-%COMP%]{display:flex;height:100%}.hidden[_nghost-%COMP%]{display:none}.tabs__panel[_ngcontent-%COMP%]{background-color:#d7d7d7;width:100%;box-shadow:inset 0px 0px 0px 1px #c9c9c9;border-radius:6px;padding:30px 15px}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TabComponent, [{
        type: Component,
        args: [{
                selector: "tab",
                host: {
                    "[class.hidden]": "!active"
                },
                template: "\n    <div class=\"tabs__panel\">\n      <ng-content></ng-content>\n    </div>\n  ",
                styles: ["\n    *{margin:0px;padding:0px}:host{display:flex;height:100%}:host(.hidden){display:none}.tabs__panel{background-color:#d7d7d7;width:100%;box-shadow:inset 0px 0px 0px 1px #c9c9c9;border-radius:6px;padding:30px 15px}\n  "]
            }]
    }], function () { return [{ type: ɵngcc1.TabsContainer }]; }, { active: [{
            type: Input
        }], tabTitle: [{
            type: Input
        }] }); })();
    return TabComponent;
}());
export { TabComponent };
